### DISCUSSION TITLE
Is there a way to link object files into a library?

### DISCUSSION THREAD

**Discussion by Pokeman2003** [vhelin/wla-dx#629] [Q&A] [answered]
Let's say I have 3 files that I want to turn into one .lib file. Presumably, they're object files, although they could also be library files. Is there any way to do that with the linker rather than just doing a series of hacky .INCLUDEs? I spent a while in the manual and couldn't find anything like it.

**Comment by cr1901** [comment, marked as answer]
AFAIK, a single assembly file maps to a single `.lib` file, just like a single assembly file maps to a single `.obj` file. So why would you use `.lib`? Because you don't have to specify as much info (`MEMORYMAP`) about your target.

I think concat'ing libs into a single file would be interesting, but it's probably subject to a bunch of design decisions that Ville doesn't want to deal with :P.

**Reply by Pokeman2003** [reply]
Disappointing. Oh well, I'll just make a script which merges the files together.

**Reply by vhelin** [reply]
Yeah, currently there is no support for what is asked here, but maybe someone wants to create a merge tool? :) One cannot just join two binaries here together and expect it to work...

**Reply by vhelin** [reply]
And yes, I'm not planning on changing the file formats so that raw binary joins would work as I see the need for the feature asked here to be quite a niche feature. But if lots of people need this then it's another thing...

**Reply by Pokeman2003** [reply]
I'll just incorporate a solution into my current maker script. My solution's pretty simple. I'll just create a mega file based on the source code's contents, and then assemble that.
